Test/Utils/ContrailAPI/ConfigureDNS.Tests.ps1

```diff
@@ -0,0 +1,132 @@
+    }
+
+    It 'adds and removes DNS server records' {
+        $ServerUUID = Add-ContrailDNSServer -ContrailUrl $ContrailUrl -AuthToken $AuthToken `
+            -DNSServerName "CreatedByPS1Script"
+
+        $Record1Data = [VirtualDNSRecordData]::new("host1", "1.2.3.4", "A")
+        $Record1UUID = Add-ContrailDNSRecord -ContrailUrl $ContrailUrl -AuthToken $AuthToken `
+            -DNSServerName "CreatedByPS1Script" -VirtualDNSRecordData $Record1Data
+
+        $Record2Data = [VirtualDNSRecordData]::new("host2", "1.2.3.5", "A")
+        $Record2UUID = Add-ContrailDNSRecord -ContrailUrl $ContrailUrl -AuthToken $AuthToken `
+            -DNSServerName "CreatedByPS1Script" -VirtualDNSRecordData $Record2Data
+
+        Remove-ContrailDNSRecord -ContrailUrl $ContrailUrl -AuthToken $AuthToken -DNSRecordUuid $Record1UUID
+        Remove-ContrailDNSRecord -ContrailUrl $ContrailUrl -AuthToken $AuthToken -DNSRecordUuid $Record2UUID
+
+        Remove-ContrailDNSServer -ContrailUrl $ContrailUrl -AuthToken $AuthToken -DNSServerUuid $ServerUUID
+    }
+
+    It 'breaks when wrong dns mode specified' {
+        { Set-IpamDNSMode -ContrailUrl $ContrailUrl -AuthToken $AuthToken `
+            -IpamFQName @("default-domain", "default-project", "default-network-ipam") `
+            -DNSMode 'wrongdnsmode' } | Should -Throw
+    }
+
+    It 'sets DNS mode to none' {
+        Set-IpamDNSMode -ContrailUrl $ContrailUrl -AuthToken $AuthToken `
+            -IpamFQName @("default-domain", "default-project", "default-network-ipam") `
+            -DNSMode 'none'
+    }
+
+    It 'sets DNS mode to default' {
+        Set-IpamDNSMode -ContrailUrl $ContrailUrl -AuthToken $AuthToken `
+            -IpamFQName @("default-domain", "default-project", "default-network-ipam") `
+            -DNSMode 'default-dns-server'
+    }
+
+    It 'sets DNS mode to tenant without DNS servers' {
+        Set-IpamDNSMode -ContrailUrl $ContrailUrl -AuthToken $AuthToken `
+            -IpamFQName @("default-domain", "default-project", "default-network-ipam") `
+            -DNSMode 'tenant-dns-server'
+    }
+
+    It 'sets DNS mode to tenant with DNS servers' {
+        Set-IpamDNSMode -ContrailUrl $ContrailUrl -AuthToken $AuthToken `
+            -IpamFQName @("default-domain", "default-project", "default-network-ipam") `
+            -DNSMode 'tenant-dns-server' -TenantServersIPAddresses @("1.1.1.1", "2.2.2.2")
+    }
+
+    It 'sets DNS mode to virtual' {
+        $ServerUUID = Add-ContrailDNSServer -ContrailUrl $ContrailUrl -AuthToken $AuthToken `
+            -DNSServerName "CreatedByPS1Script"
+
+        Set-IpamDNSMode -ContrailUrl $ContrailUrl -AuthToken $AuthToken `
+            -IpamFQName @("default-domain", "default-project", "default-network-ipam") `
+            -DNSMode 'virtual-dns-server' -VirtualServerName "CreatedByPS1Script"
+
+        Remove-ContrailDNSServer -ContrailUrl $ContrailUrl -AuthToken $AuthToken -DNSServerUuid $ServerUUID -Force
```

Comment:
Why do we `Add` and `Remove` `ContrailDNSServer` in this test case, but not in other which verify `Set-IpamDNSMode` function?

Reply:
This hints at lack of test case isolation http://wiki.c2.com/?UnitTestIsolation

Reply:
Because other modes doesn't use Contrail DNS Servers. This one does so we need to create one.